src/connection-manager/agreement.ts

```diff
@@ -169,13 +247,17 @@ class Agreement {
     if (!hashes.has(validation.ledger_hash)) {
       hashes.set(validation.ledger_hash, Date.now())
       this.validationsByPublicKey.set(signing_key, hashes)
+      const server_version =
+        preceedFlagLedger(validation.ledger_index) &&
+        decodeServerVersion(validation.server_version)
       const validator = {
         master_key: validation.master_key,
         signing_key,
         ledger_hash: validation.ledger_hash,
         current_index: Number(validation.ledger_index),
         partial: !validation.full,
         last_ledger_time: new Date(),
+        ...(server_version && { server_version }),
```

Comment:
This seems a little complicated... 

What type are we trying to do here. Should this be type `string | null`?

Reply:
If it's type string or null, just pass in `decodedVersion`